src/system/string.rs

```diff
@@ -62,3 +76,64 @@ impl<'a> SfStrConv for &'a String {
         str.with_as_sfstr(fun)
     }
 }
+
+/// Errors which can occur when attempting to translate from UTF-32 to UTF-8
+#[derive(Debug)]
+pub struct SfStrConvError {
+    index: usize,
+    inner: Option<Vec<u32>>,
+}
+
+impl SfStrConvError {
+    /// Returns the index in the string where the conversion error occured
+    pub fn index(&self) -> usize {
+        self.index
+    }
+
+    /// Returns the invalid code point that failed to convert to UTF-8 in UTF-32 form
+    ///
+    /// If the sequence that generated the error was a reference to a slice instead of a [`Vec`],
+    /// this will return [`None`].
+    pub fn invalid_code_point(&self) -> Option<u32> {
+        self.inner
+            .as_ref()
+            .and_then(|inner| inner.get(self.index).copied())
+    }
+
+    /// Returns the underlying vector of values which generated the error in the first place.
+    ///
+    /// If the sequence that generated the error was a reference to a slice instead of a [`Vec`],
+    /// this will return [`None`].
+    #[inline]
+    #[cfg_attr(docsrs, doc(cfg(feature = "alloc")))]
+    #[must_use]
+    pub fn into_vec(self) -> Option<Vec<u32>> {
+        self.inner
+    }
+}
+
+impl From<Utf32Error> for SfStrConvError {
```

Comment:
Instead of the From impl, it could just wrap Utf32Error, since it has private fields anyway

Reply:
Could you show me what you mean? 

Also, I see nothing wrong with the from implementation